tests/parser/persistentPageIdentifierFunction.txt

```diff
@@ -0,0 +1,59 @@
+!! options
+parsoid-compatible
+version=2
+!! end
+
+# Force the test runner to ensure the extension is loaded
+!! functionhooks
+ppid
+!! endfunctionhooks
+
+!! test
+Returns nothing without a page context
+!! wikitext
+{{#ppid:}}
+!! html
+!! end
+
+!! article
+Test Page
+!! text
+!! endarticle
+
+!! test
+Returns persistent identifier for page
+!! config
+wgPersistentPageIdentifiersParserTestStubId="test-42"
+!! options
+title=[[Test Page]]
+!! wikitext
+{{#ppid:}}
+!! html
+<p>test-42
+</p>
+!! end
+
+!! test
+Returns formatted persistent identifier for page
+!! config
+wgPersistentPageIdentifiersParserTestStubId="test-43"
+wgPersistentPageIdentifiersFormat="foo/$1/bar"
+!! options
+title=[[Test Page]]
+!! wikitext
+{{#ppid:}}
+!! html
+<p>foo/test-43/bar
+</p>
+!! end
+
+!! test
+Returns nothing when page does not have a persistent identifier
```

Comment:
This is not the same as `Returns nothing without a page context`. That test makes sure the parser function doesn't blow up if it gets called in a weird context.
This test is to simulate an existing page before the extension was enabled.